MHRA/products
# Timlee/add pars logging

# Add logging for when an SPC/PIL/PAR is uploaded or deleted

- Log successful uploads and deletes of all file types to an append blob in Azure storage
- Add `transaction-log-file-creator` to create a new append blob log file at the start of every month (otherwise file risks getting too large, leading to data loss) - triggered by cron workflow

Addresses #775 

![](https://media.giphy.com/media/XCLBNof6ICAEM/giphy.gif)

### Acceptance Criteria
See ticket

### Testing information
- Execute `transaction-log-file-creator` locally with `cargo run` or `make` to see new file created in Azure storage
- Run locally, create or delete a file and observe log file being written to in Azure storage

### Pre-flight Checklist


### Pre-merge Checklist


[testing strategy]: https://github.com/MHRA/products/blob/master/docs/principles/testing.md "MHRA/products testing strategy"
